### ISSUE TITLE
Attributes of Arc Initialized Objects Not Detected

### ISSUE THREAD

**Issue by fredfortier** [intellij-rust/intellij-rust#3879]
## Environment

* **Intellij-Rust plugin version: v0.2.97.2123-191** 
* **Rust toolchain version: rustc 1.36.0-nightly (3991285f5 2019-04-25)**
* **IDE name and version: IntelliJ Ultimate 2019.1.2**
* **Operating system: GNU/Linux 5.0.16-300.fc30.x86_64**

## Problem description

Attributes and methods of Struct instances initialized with `Arc` are not recognized (no autocomplete nor code navigation). This is a new problem for me. I'm not sure if it's correlated to a Rust plug-in upgrade or a Rust toolchain upgrade since I did both recently. I must use this particular toolchain version.

I tried re-initializing caches and deleting `~/.cargo` completely.
 
## Steps to reproduce

```rust
struct Hello {
   foo: String,
   bar: String,
}
let hello = Arc::new(Hello::default());
hello.fo //<- CTRL+SPACE should list `foo` but it does not. It only lists generic methods, nothing specific to `Hello`.
```

**Comment by vlad20012** [comment]
At your example `#[derive(Default)]` missed for `Hello` struct. I can't reproduce the issue if I add the derive.

**Comment by parisholley** [comment]
This is an issue outside structs as well... running into this with default implemenation of mongodb crate:

https://docs.rs/mongodb/0.4.0/mongodb/index.html

The library exposes an API like this:

```rust
pub type Client = Arc<ClientInner>;
```

Yet calling the code like so yields no intellisense:

```rust
let client: ThreadedClient = Client::with_uri(...)
```

**Comment by Undin** [comment]
@parisholley your issue is not related to the original one.
Yours should be fixed by #4936
